From: Takashi Iwai <tiwai@suse.de>
Subject: Fix arm64 breakage with this_cput_write()
Patch-mainline: Never, a temporary quick fix
References: bsc#1089343 CVE-2018-3646

The patch patches.arch/17-cpu-hotplug-Boot-HT-siblings-at-least-once.patch
replaced the original per_cpu_ptr() code with this_cpu_write().
While it should work on all architecture, arm64 seems misbehaving and
misses the offset by some reason.

As a quick workaround, make the code back to per_cpu_ptr() reference.

Suggested-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 kernel/cpu.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/kernel/cpu.c
+++ b/kernel/cpu.c
@@ -2056,8 +2056,8 @@ void __init boot_cpu_init(void)
  */
 void __init boot_cpu_state_init(void)
 {
-	this_cpu_write(cpuhp_state.booted_once, true);
-	this_cpu_write(cpuhp_state.state, CPUHP_ONLINE);
+	per_cpu_ptr(&cpuhp_state, smp_processor_id())->booted_once = true;
+	per_cpu_ptr(&cpuhp_state, smp_processor_id())->state = CPUHP_ONLINE;
 }
 
 /* kabi */
